modulus/models/graphcast/graph_cast_processor.py

```diff
@@ -177,3 +179,50 @@ def forward(
             )
 
         return efeat, nfeat
+
+
+class GraphCastProcessorGraphTransformer(nn.Module):
+    """Processor block used in GenCast operating on a latent space
+    represented by hierarchy of icosahedral meshes.
+
+    Parameters
+    ----------
+    """
+
+    def __init__(
+        self,
+        attn_mask,
+        processor_layers: int = 16,
+        input_dim_nodes: int = 512,
+        hidden_dim: int = 512,
+    ):
+        super().__init__()
+
+        layers = [
+            te.pytorch.TransformerLayer(
+                hidden_size=input_dim_nodes,
+                ffn_hidden_size=hidden_dim,
+                num_attention_heads=4,
+                layer_number=i + 1,
+                fuse_qkv_params=False,
+            )
+            for i in range(processor_layers)
+        ]
+
+        self.processor_layers = nn.ModuleList(layers)
+        self.hidden_dim = hidden_dim
+        self.attn_mask = torch.tensor(attn_mask, dtype=torch.bool)
+
+    def forward(
+        self,
+        nfeat: Tensor,
+    ) -> Tensor:
+        nfeat = nfeat.unsqueeze(1)
```

Comment:
Note: This follows the default "sbhd” layout.

Reply:
My understanding is that the reshape is done automatically inside the self-attention module. I get a "GEMM not possible" error if I reshape manually.